lib/MusicBrainz/Server/Data/Editor.pm

```diff
@@ -186,8 +186,13 @@ sub find_by_ip {
 sub search_by_email {
     my ($self, $email) = @_;
 
+    # Remove periods (\.), and anything preceded by a + sign for email matching,
+    # since both of these are often used as "free aliases" by email providers
+    $email =~ s/(\\.|\+[\d]+)//g;
+
     my $query = 'SELECT ' . $self->_columns .
-        ' FROM ' . $self->_table . ' WHERE email ~ ?' .
+        ' FROM ' . $self->_table .
+        ' WHERE regexp_replace(email, \'(\.|\+[\d]+)\', \'\', \'g\') ~ ?' .
```

Comment:
Same as above but in PostgreSQL:

```suggestion
        ' WHERE regexp_replace(regexp_replace(email, \'\(+.*@)\', \'@\'), \'\.(.*@)\', \'\1\', \'g\') ~ ?' .
```

Reply:
Also, let's use a different quoting operator like `q""` or `q{}` so we don't have to escape `'` 10 times. :)

Reply:
(Alternatively, you can use `?` and pass the regexp string as a parameter; not sure which would be clearer.)